CodeGeneration/Sources/generate-swiftsyntax/templates/swiftparser/ParserTokenSpecSetFile.swift

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let parserTokenSpecSetFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
+  DeclSyntax("@_spi(RawSyntax) import SwiftSyntax")
+
+  try! ExtensionDeclSyntax("extension Parser") {
+    for node in SYNTAX_NODES {
+      if let children = node.layoutNode?.children {
+        for child in children where !child.isUnexpectedNodes {
+          if case let .token(choices, _, _) = child.kind, choices.count > 1 {
+            try EnumDeclSyntax("enum \(raw: node.varOrCaseName.description.withFirstCharacterUppercased + child.name): TokenSpecSet") {
```

Comment:
Just an idea: What do you think about generating something like

```swift
extension AccessorDecl {
  public enum AccessorKindOptions: TokenSpecSet {
  }
}
```

---

Instead of `node.varOrCaseName.description.withFirstCharacterUppercased`, you should use `self.kind.syntaxType`.